src/deepsparse/transformers/utils/token_generator.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+import numpy
+
+from deepsparse.utils.data import numpy_softmax
+
+
+class TokenGenerator:
+    def __init__(
+        self,
+        logits: numpy.ndarray,
+        tokens: List[int] = [],
+        deterministic: bool = True,
+        sampling_temperature: float = 1.0,
+        top_k: int = 0,
+        top_p: float = 0.0,
+        frequency_penalty: float = 0.0,
+        presence_penalty: float = 0.0,
+        **kwargs,
+    ):
+        self.token_frequencies = numpy.zeros(logits.shape[-1])
+
+        self.deterministic = deterministic
+        self.sampling_termperature = sampling_temperature
+        self.top_k = top_k
+        self.top_p = top_p
+        self.frequency_penalty = frequency_penalty
+        self.presence_penalty = presence_penalty
+        self.tokens = []
+        for token in tokens:
+            self.update_frequencies(token)
```

Comment:
probably not the most expensive, but you can rewrite using `numpy.unique(tokens, return_counts=True)` to vectorize

Reply:
```
tokens = [1,2,3,4,5]
>>> numpy.unique(tokens, return_counts=True)
(array([1, 2, 3, 4, 5]), array([1, 1, 1, 1, 1]))
```

and making it True will miss duplicates